pre_commit/clientlib.py

```diff
@@ -144,18 +144,13 @@ def check(self, dct: Dict[str, Any]) -> None:
                 f"regex, not a glob -- matching '/*' probably isn't what you "
                 f'want here',
             )
-        if r'[\/]' in dct.get(self.key, ''):
-            logger.warning(
-                fr'pre-commit normalizes slashes in the {self.key!r} field '
-                fr'in hook {dct.get("id")!r} to forward slashes, so you '
-                fr'can use / instead of [\/]',
-            )
-        if r'[/\\]' in dct.get(self.key, ''):
-            logger.warning(
-                fr'pre-commit normalizes slashes in the {self.key!r} field '
-                fr'in hook {dct.get("id")!r} to forward slashes, so you '
-                fr'can use / instead of [/\\]',
-            )
+        for fwd_slash_re in [r'[\\/]', r'[\/]', r'[/\\]']:
```

Comment:
Also: should `r'[\/]'` trigger the warning at all? The matched character _is_ a forward slash, mentioning "normalization to forward slashes" seems odd in this case.

Reply:
This is something I expected you to comment on, and I'm glad for that. 

Since this was accepted in the [previous pull request regarding these regexes](https://github.com/pre-commit/pre-commit/pull/2053#issuecomment-983436239), without any comments on the subject, I felt like confirming if this is meant to be a style guideline?
In this case, the warning would (sort of) make sense.

Reply:
Hello, and thanks for working on this.

Yes, you’re right that pre-commit could warn about `r'[\/]'` as a suggestion about style (I assume the issue didn’t come up in the original PR for an oversight).
But in that case, I’d be more inclined to extract the check into a separate one, with its own proper message.

I understand that this could be quite a nit pick, and that grouping all the regexes under the same message would be equally acceptable, although a little less precise with regards to semantics.

@asottile any thoughts about this?

Reply:
yeah so admittedly the original task was a _mistake_ on my part since `r'[\/]'` is the same as `r'[/]'` -- but since there's already several levels of where escaping is questionable (yaml rules, then python rules, then regex rules) I think it's still correct to warn about that

the flip `r'[/\]'` doesn't need a warning because it's an invalid regex (unclosed `]`) so something will already let the user know something is broken